test/functional/test_framework/mininode.py

```diff
@@ -367,7 +367,7 @@ def on_version(self, message):
 
     def wait_for_disconnect(self, timeout=60):
         test_function = lambda: not self.is_connected
-        wait_until(test_function, timeout=timeout, lock=mininode_lock)
+        wait_until(test_function, timeout=timeout, lock=mininode_lock, factor=self.factor)
```

Comment:
Given that they are all based on `def wait_until` wouldn't it make more sense to treat `wait_until` as the basic mechanism for timed waits and only multiply it in there?